pandas/core/groupby/groupby.py

```diff
@@ -1774,22 +1774,13 @@ def _bool_agg(self, val_test: Literal["any", "all"], skipna: bool):
         """
 
         def objs_to_bool(vals: ArrayLike) -> tuple[np.ndarray, type]:
-            if is_object_dtype(vals.dtype):
+            if is_object_dtype(vals.dtype) and skipna:
                 # GH#37501: don't raise on pd.NA when skipna=True
-                if skipna:
-                    func = np.vectorize(
-                        lambda x: bool(x) if not isna(x) else True, otypes=[bool]
-                    )
-                    vals = func(vals)
-                else:
-                    vals = vals.astype(bool, copy=False)
-
-                vals = cast(np.ndarray, vals)
+                vals = vals.copy()
+                vals[isna(vals)] = True
```

Comment:
Can you add a copy that we will re-compute a mask later on? Hard to follow why setting = True is valid if you don't know that.

Additionally, would it be faster to check if isna(vale).any() before making a copy?

Reply:
Added a comment and skipped the copy when possible. Seems to save a bit more more (~5%).